crates/ty_project/src/lib.rs

```diff
@@ -523,9 +540,12 @@ impl Project {
             }
         }
 
-        if self
-            .open_fileset(db)
-            .is_none_or(|files| !files.contains(&file))
+        // Avoid dropping the AST if the check mode is `AllFiles` because otherwise the next check
+        // request for the same mode will need to re-parse the file again.
+        if matches!(self.check_mode(db), CheckMode::OpenFiles)
```

Comment:
We do want to drop AST nodes even in `CheckMode::AllFiles` if they're not open. Reparsing should only be necessary if a dependency of that file changed (and only if that files public interface changed).

Reply:
https://github.com/astral-sh/ruff/pull/19255